include/boost/wintls/detail/sspi_handshake.hpp

```diff
@@ -70,7 +76,7 @@ class sspi_handshake {
         case handshake_type::server:
           return SECPKG_CRED_INBOUND;
       }
-      BOOST_UNREACHABLE_RETURN(0);
+      std::abort(); // unreachable here
```

Comment:
This doesn't really do the same as the boost macro.

Might make sense to move this in another header that calls the correct builtin for various compilers. Maybe the same place as the assert macro?

I'm not a huge of generic names like `util.hpp` but in this case it might be acceptable.

Alternatively add separate headers for various compilers included from `config.hpp` similar to what boost does.